bin/anthology/volumes.py

```diff
@@ -99,9 +99,8 @@ def _set_meta_info(self, meta_data):
             self.attrib["editor"] = self.attrib["author"]
             del self.attrib["author"]
 
-        # Expand URL if present
-        if "url" in self.attrib:
-            self.attrib["url"] = infer_url(self.attrib["url"])
+        # Expand URL if present, or create URL field if not
+        self.attrib["url"] = infer_url(self.attrib.get("url", self.full_id))
```

Comment:
When would the “url” attrib *not* be `self.full_id`? Can we comment or explain that here?

I am trying to remember: I think that `<url>` is not always set in the XML, but that is used to point to the PDF, which is sometimes offsite. In the code, we use “url” for the canonical page. Isn’t that always just the page inside the Anthology?

Reply:
For offsite URLs in `<url>`, both "url" and "pdf" will currently point to the same offsite URL. (That's legacy behaviour.)

Offsite URLs currently never occur within a volume's `<meta>` block, so `self.attrib["url"]` here will indeed always be equal to `self.full_id` whenever it is set...

Reply:
I'll merge this now as is, as the code for getting the URL changes anyway in #1473, and I'd mainly like to merge this so I can properly compare the two builds :)